aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 2:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0xFFFFFFFF, 0x0, 0x0};
+          vec.values.val[0] = (float32x4_t)mask_low;
+          vec.values.val[1] = a.values.val[1];
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 3:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = (float32x4_t)mask_low;
+          vec.values.val[1] = a.values.val[1];
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 4:
+        return Vec256<float>(b.values.val[0], a.values.val[1]);
+      case 5:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0x0, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 6:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 7:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+    }
+    return b;
+  }
+  static Vec256<float> loadu(const void* ptr, int64_t count = size()) {
+    if (count == size()) {
+      volatile register float32x4_t low asm("q3");
```

Comment:
The [register keyword is deprecated in C++](https://en.cppreference.com/w/cpp/keyword/register) and will be a compile-time error in C++17 once PyTorch switches over.

Reply:
Yes, but without that the inline assembly code gets longer as I have to introduce move instructions inside to move data to d6/d7. And since load/stores are commonly used ones it will have perf implicaiton.

Reply:
Well what this means is that we are only postponing this issue since whoever will port PyTorch to C++-17 in a year or two will have to deal with this. :)  Anyhow I am OK with that if you and David are.

Reply:
This is true, but hopefully clang will fix the issue by then and android ndk will be updated with it, so we can move away from this stop gap measure.